feat: avatars & account menu overhaul

reneaaron opened the pull request:
### Describe the changes you have made in this PR

![image](https://user-images.githubusercontent.com/100827540/224453985-9b107f31-6731-40a3-a35b-7cb2665ee98b.png)

stackingsaunter commented:
I prepared handoff for the dropdown 

<img width="637" alt="Screenshot 2023-03-17 at 23 04 52" src="https://user-images.githubusercontent.com/68239231/226061140-455fe59c-d30d-47ab-94a7-3066d5bfe418.png">

<img width="1155" alt="Screenshot 2023-03-17 at 23 05 05" src="https://user-images.githubusercontent.com/68239231/226061167-939f833a-f037-4575-9c35-b8ea1f3f8109.png">

[Figma file here ](https://www.figma.com/file/AMzMvcwrr2IdunqMakaB8r/navigation%2C-dropdown%2C-receive-onchain-work%2C-connectors%2C-nostr-events?node-id=119-31384&t=PLf4EExSybimRo3k-0)

stackingsaunter commented:
Dark version uses white text and neutral300 for "balance" and fiat
<img width="367" alt="Screenshot 2023-03-18 at 18 46 16" src="https://user-images.githubusercontent.com/68239231/226124273-ce256844-219c-418d-835a-f76e7f40e0ae.png">

bitcoinuser commented:
> Dark version uses white text and neutral300 for "balance" and fiat <img alt="Screenshot 2023-03-18 at 18 46 16" width="367" src="https://user-images.githubusercontent.com/68239231/226124273-ce256844-219c-418d-835a-f76e7f40e0ae.png">

Hi, I think instead of (Switch Account) should be (Selected Account). And in the selected account itens could list the one that is selected with the check mark icon in front, the way works today.

reneaaron commented:
There are still some open todos in here, but maybe we can do a first round of review? 

**📃 Open tasks**
- [ ] Fix tests
- [ ] Fix overflow for long account names
- [ ] Design review

reneaaron commented on src/app/components/AccountMenu/index.tsx:
```diff
@@ -73,49 +78,77 @@ function AccountMenu({ showOptions = true }: Props) {
+            <WalletIcon className="h-5 w-5 mr-2 text-gray-700 dark:text-neutral-300" />
+            {t("options.account.account_settings")}
+          </Menu.ItemButton>
+          {isAlbyAccount(authAccount?.alias) && (
```
How could we get rid of this? Currently we don't access the initialized connector in here 🤔 (and even with the connector we would need to introduce an `Alby` connector (+ try to migrate older ones?)

bumi replied:
we only have the alias in the authAccount not the host or something else? 

guess we should move asap to an individual connector to have a proper way to identify this.
right now this could be as easy as duplicating lndhub.ts into alby.ts and setting the connector to `alby`.

reneaaron replied:
Added some info to #1956

rolznz commented:
In the new design you cannot see your balance without clicking on the account select dropdown
![image](https://user-images.githubusercontent.com/33993199/229995991-c6a26465-fb14-4e59-b45a-7e9cedb7309d.png)

I think we should keep the balance always visible. @stackingsaunter @reneaaron thoughts?

rolznz reviewed (approved):
tACK

rolznz commented:
@reneaaron @stackingsaunter great improvement. The menu looks much more clean and the links to manage your Alby account should reduce a lot of confusion.

bumi commented on src/app/components/AccountMenu/index.tsx:
```diff
@@ -73,49 +77,78 @@ function AccountMenu({ showOptions = true }: Props) {
+            {t("options.account.account_settings")}
+          </Menu.ItemButton>
+          {isAlbyAccount(authAccount?.alias) && (
+            <Menu.ItemButton
```
I am wondering if we should add this also (or only) to the account detail page. 
Right now we say "Account settings" and "manage the Alby account" 
maybe it can only be "Account settings" and in there we have a prominent link to manage the Alby account? 

potentially this could even be its own section explaining the user a bit the difference?

bumi commented on src/app/components/AccountMenu/index.tsx:
```diff
@@ -73,49 +77,78 @@ function AccountMenu({ showOptions = true }: Props) {
+          {isAlbyAccount(authAccount?.alias) && (
+            <Menu.ItemButton
+              onClick={() => {
+                window.open(`https://getalby.com/login`, "_blank");
```
I always link to the /user  - even though we redirect from the login page.

bumi reviewed (approved):
this looks great! nice work! 

I did not check the code in detail, but LGTM.

bumi commented:
![CleanShot 2023-04-05 at 10 30 09@2x](https://user-images.githubusercontent.com/318/230026055-5e1931bb-eecc-457e-b9e4-0c32c4d2a9b8.png)

is this link always going to the accounts list? because now if I open the accounts details from the new menu, then this back link does not work (there is nothing to go back to)

stackingsaunter commented:
> I think we should keep the balance always visible. @stackingsaunter @reneaaron thoughts?

I think it's important on the main screen and when transacting, that's why I think it's better to move account balance to the amount inputs, see here https://github.com/getAlby/lightning-browser-extension/issues/2237 and https://github.com/getAlby/lightning-browser-extension/issues/2236. It should work well if those all are implemented

stackingsaunter commented:
If there's no account to switch, can we not have "switch account" in the dropdown? 

![Screenshot 2023-04-05 at 16 43 55](https://user-images.githubusercontent.com/68239231/230116784-6da4c30b-370d-4334-bb4a-10bb9ae58678.png)

reneaaron commented:
> guess we should move asap to an individual connector to have a proper way to identify this.
> right now this could be as easy as duplicating lndhub.ts into alby.ts and setting the connector to alby.

Added those things and some todos to #1956 and added it to our project board, let's tackle that in a follow up PR.

Tackled the feedback from this PR, thanks for testing! 🙌
